src/hotspot/os/windows/os_windows.cpp

```diff
@@ -3216,51 +3216,77 @@ static char* allocate_pages_individually(size_t bytes, char* addr, DWORD flags,
   return p_buf;
 }
 
-static size_t large_page_init_decide_size() {
+size_t os::large_page_init_decide_size() {
   // print a warning if any large page related flag is specified on command line
   bool warn_on_failure = !FLAG_IS_DEFAULT(UseLargePages) ||
-                         !FLAG_IS_DEFAULT(LargePageSizeInBytes);
+      !FLAG_IS_DEFAULT(LargePageSizeInBytes);
 
 #define WARN(msg) if (warn_on_failure) { warning(msg); }
+#define WARN1(msg,p) if (warn_on_failure) { warning(msg,p); }
 
   if (!request_lock_memory_privilege()) {
-    WARN("JVM cannot use large page memory because it does not have enough privilege to lock pages in memory.");
-    return 0;
+      WARN("JVM cannot use large page memory because it does not have enough privilege to lock pages in memory.");
+      return 0;
   }
 
-  size_t size = GetLargePageMinimum();
+size_t size = GetLargePageMinimum();
   if (size == 0) {
-    WARN("Large page is not supported by the processor.");
-    return 0;
+      WARN("Large page is not supported by the processor.");
+      return 0;
   }
 
-#if defined(IA32) || defined(AMD64)
-  if (size > 4*M || LargePageSizeInBytes > 4*M) {
-    WARN("JVM cannot use large pages bigger than 4mb.");
-    return 0;
-  }
+  if (!os::win32::is_windows_11_or_greater() && !os::win32::is_windows_server_2022_or_greater()) {
+#if defined(IA32)
+      if (size > 4 * M || LargePageSizeInBytes > 4 * M) {
+          WARN("JVM cannot use large pages bigger than 4mb.");
+          return 0;
+      }
+#elif defined(AMD64)
+      if (!EnableAllLargePageSizes) {
+          if (size > 4 * M || LargePageSizeInBytes > 4 * M) {
+              WARN("JVM cannot use large pages bigger than 4mb.");
+              return 0;
+          }
+      }else {
+          WARN("EnableAllLargePageSizes flag is ignored on Windows versions prior to Client 11 and Server 22 due to limited support.");
+      }
 #endif
+      }
 
   if (LargePageSizeInBytes > 0 && LargePageSizeInBytes % size == 0) {
-    size = LargePageSizeInBytes;
+      size = LargePageSizeInBytes;
+  }else {
```

Comment:
Put spaces on both sides of control flow keywords if, else, for, switch, etc.
Source: [HotSpot Coding Style](https://github.com/openjdk/jdk/blob/master/doc/hotspot-style.md)